lib/takeout/response.rb

```diff
@@ -0,0 +1,28 @@
+module Takeout
+  class Response
+    attr_accessor :headers
+    attr_accessor :body
+    attr_accessor :response
+    attr_accessor :options
+
+    def initialize(options={})
+      if block_given?
+        yield self
+      else
+        extract_instance_variables_from_options(options)
+      end
+    end
+
+    def extract_instance_variables_from_options(options)
+      # Set instance variables
+      @headers = options[:headers] || ''
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.